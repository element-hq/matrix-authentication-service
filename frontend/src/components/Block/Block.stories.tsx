// Copyright 2024 New Vector Ltd.
// Copyright 2023, 2024 The Matrix.org Foundation C.I.C.
//
// SPDX-License-Identifier: AGPL-3.0-only
// Please see LICENSE in the repository root for full details.

import type { Meta, StoryObj } from "@storybook/react";
import { Body, H1, H5 } from "@vector-im/compound-web";

import Block from "./Block";

const meta = {
  title: "UI/Block",
  component: Block,
  tags: ["autodocs"],
} satisfies Meta<typeof Block>;

export default meta;
type Story = StoryObj<typeof Block>;

export const Basic: Story = {
  render: (args) => (
    <Block {...args}>
      <H1>Title</H1>
      <H5>Subtitle</H5>
      <Body justified>
        Lorem ipsum dolor sit amet, officia excepteur ex fugiat reprehenderit
        enim labore culpa sint ad nisi Lorem pariatur mollit ex esse
        exercitation amet. Nisi anim cupidatat excepteur officia. Reprehenderit
        nostrud nostrud ipsum Lorem est aliquip amet voluptate voluptate dolor
        minim nulla est proident. Nostrud officia pariatur ut officia. Sit irure
        elit esse ea nulla sunt ex occaecat reprehenderit commodo officia dolor
        Lorem duis laboris cupidatat officia voluptate. Culpa proident
        adipisicing id nulla nisi laboris ex in Lorem sunt duis officia eiusmod.
        Aliqua reprehenderit commodo ex non excepteur duis sunt velit enim.
        Voluptate laboris sint cupidatat ullamco ut ea consectetur et est culpa
        et culpa duis.
      </Body>
    </Block>
  ),
};
