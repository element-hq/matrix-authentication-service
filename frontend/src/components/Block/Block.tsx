// Copyright 2024 New Vector Ltd.
// Copyright 2022-2024 The Matrix.org Foundation C.I.C.
//
// SPDX-License-Identifier: AGPL-3.0-only
// Please see LICENSE in the repository root for full details.

import { Heading } from "@vector-im/compound-web";
import cx from "classnames";
import type { ReactNode } from "react";

import styles from "./Block.module.css";

type Props = React.PropsWithChildren<{
  title?: ReactNode;
  className?: string;
  highlight?: boolean;
}>;

const Block: React.FC<Props> = ({ children, className, highlight, title }) => {
  return (
    <div className={cx(styles.block, className)} data-active={highlight}>
      {title && (
        <Heading as="h4" size="sm" weight="semibold" className={styles.title}>
          {title}
        </Heading>
      )}

      {children}
    </div>
  );
};

export default Block;
