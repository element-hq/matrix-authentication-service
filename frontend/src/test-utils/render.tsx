// Copyright 2024 New Vector Ltd.
//
// SPDX-License-Identifier: AGPL-3.0-only
// Please see LICENSE in the repository root for full details.

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render as testingLibraryRender } from "@testing-library/react";
import { DummyRouter } from "./router";

const client = new QueryClient();

const render = (
  ui: Parameters<typeof testingLibraryRender>[0],
  options: Parameters<typeof testingLibraryRender>[1] = {},
): ReturnType<typeof testingLibraryRender> =>
  testingLibraryRender(ui, {
    wrapper: ({ children }) => (
      <QueryClientProvider client={client}>
        <DummyRouter>{children}</DummyRouter>
      </QueryClientProvider>
    ),
    ...options,
  });

export default render;
