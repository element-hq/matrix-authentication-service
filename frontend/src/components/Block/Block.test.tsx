// Copyright 2024 New Vector Ltd.
// Copyright 2023, 2024 The Matrix.org Foundation C.I.C.
//
// SPDX-License-Identifier: AGPL-3.0-only
// Please see LICENSE in the repository root for full details.

// @vitest-environment happy-dom

import { describe, expect, it } from "vitest";

import render from "../../test-utils/render";
import Block from "./Block";

describe("Block", () => {
  it("render <Block />", () => {
    const { asFragment } = render(<Block />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("render <Block /> with children", () => {
    const { asFragment } = render(
      <Block>
        <h1>Title</h1>
        <p>Body</p>
      </Block>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("passes down the className prop", () => {
    const { asFragment } = render(<Block className="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with highlight", () => {
    const { asFragment } = render(<Block highlight />);
    expect(asFragment()).toMatchSnapshot();
  });
});
