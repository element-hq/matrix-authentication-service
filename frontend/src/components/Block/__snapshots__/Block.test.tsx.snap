// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Block > passes down the className prop 1`] = `
<DocumentFragment>
  <div
    class="_block_17898c test"
  />
</DocumentFragment>
`;

exports[`Block > render <Block /> 1`] = `
<DocumentFragment>
  <div
    class="_block_17898c"
  />
</DocumentFragment>
`;

exports[`Block > render <Block /> with children 1`] = `
<DocumentFragment>
  <div
    class="_block_17898c"
  >
    <h1>
      Title
    </h1>
    <p>
      Body
    </p>
  </div>
</DocumentFragment>
`;

exports[`Block > renders with highlight 1`] = `
<DocumentFragment>
  <div
    class="_block_17898c"
    data-active="true"
  />
</DocumentFragment>
`;
